interactions/models/internal/context.py

```diff
@@ -346,7 +346,7 @@ def author_permissions(self) -> Permissions:
 
     @property
     def command(self) -> InteractionCommand:
-        return self.client._interaction_lookup[self._command_name]
+        return self.client._interaction_lookup.get(self._command_name)
```

Comment:
This does make this breaking, right? I'm going to let another contributor look over this to give their own judgment, but this definitely changes how people should interface with the property.

Reply:
Not necessarily. If the command doesn't exist it will throw a `KeyError` whilst this will just return None.

The logic is already in place to handle as if it's supposed to return None.

See:

https://github.com/interactions-py/interactions.py/blob/2a896b7a34a42bf4abe63b6a7b4af17a1a60b722/interactions/client/client.py#L1913-L1951

```python
if ctx.command:
                self.logger.debug(f"{ctx.command_id}::{ctx.command.name} should be called")
```
but this never is not true because it will return a KeyError before it reaches there.

Reply:
I saw that too, but that doesn't mean changing the property still wouldn't be breaking. It just points out how this is not how it was meant to be designed, but I know I've been told before that fixes to that would still be breaking.

Again, I'm going to let another contributor decide on a judgment since I don't have the best judgment on what's breaking or not.

Reply:
You are right in the sense that this change could be breaking. I'm sure people have also written counter logic from the errors that pertain from this issue. The thing about this issue is that it's impossible to try-catch this error since it comes from the an independent callback - to my knowledge. However, there does need to be a proper resolution.

Reply:
I'm a bit confused about not being able to try catch the error - you should absolutely be able to. If you mean how the command dispatcher will error... well, yes, you can't really error handle things like that from a bot development POV.

That being said, I'm pretty sure you could easily modify the library itself to make the interaction command dispatcher use a try catch itself.

Reply:
Not necessarily - it can happen if your commands are out of sync in general, which... is also undefined, granted.

I suppose the only remaining question then is if the typehint needs to be changed.

Reply:
Yeah, definitely needs to be marked as Optional

Reply:
Thanks for feedback. If you all have anymore feedback please let me know.